Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -293,6 +281,35 @@ lemma indep_iff_forall_subset_not_circuit (hI : I ⊆ M.E := by aesop_mat) :
     M.Indep I ↔ ∀ C, C ⊆ I → ¬M.Circuit C := by
   rw [indep_iff_forall_subset_not_circuit', and_iff_left hI]
 
+/-! ### Closure -/
+
+lemma Circuit.closure_diff_singleton_eq (hC : M.Circuit C) (e : α) :
+    M.closure (C \ {e}) = M.closure C :=
+  (em (e ∈ C)).elim
+    (fun he ↦ by rw [(hC.diff_singleton_basis he).closure_eq_closure])
+    (fun he ↦ by rw [diff_singleton_eq_self he])
+
+lemma Circuit.subset_closure_diff_singleton (hC : M.Circuit C) (e : α) :
+    C ⊆ M.closure (C \ {e}) := by
+  rw [hC.closure_diff_singleton_eq]
+  exact M.subset_closure _ hC.subset_ground
+
+lemma Circuit.mem_closure_diff_singleton_of_mem (hC : M.Circuit C) (heC : e ∈ C) :
+    e ∈ M.closure (C \ {e}) :=
+  (hC.subset_closure_diff_singleton e) heC
```

Comment:
```suggestion
  hC.subset_closure_diff_singleton e heC
```

Reply:
Actually now that I look at it more closely, since this is just a combination of `heC` with `Circuit.subset_closure_diff_singleton`, does this lemma need to exist?

Reply:
It gets a bit of use in the repo, and its use reduces defeq abuse by epsilon, so I prefer to have it around.

Reply:
I'm not quite certain epsilon is nonzero here, but let's go with it for now!